src/http/one/Parser.cc

```diff
@@ -273,8 +283,11 @@ Http::One::ErrorLevel()
 
 // BWS = *( SP / HTAB ) ; WhitespaceCharacters() may relax this RFC 7230 rule
 bool
-Http::One::ParseBws(Tokenizer &tok)
+Http::One::ParseBws(Parser::Tokenizer &tok)
 {
+    if (tok.atEnd())
+        return false; // need more data
+
     if (const auto count = tok.skipAll(Parser::WhitespaceCharacters())) {
         // Generating BWS is a MUST-level violation so warn about it as needed.
         debugs(33, ErrorLevel(), "found " << count << " BWS octets");
```

Comment:
This function remains buggy: We should not return true after skipping a few BWS characters unless a non-BWS character follows those skipped characters.

tok.skipAll() is very similar to tok.prefix(); its callers need the same careful treatment of partial values.

Reply:
> similar to tok.prefix()

tok.prefix() may return true after skipping a few permitted characters without following a non-permitted one. Please see its implementation (when "whole haystack matched").

Reply:
> tok.prefix() may return true after skipping a few permitted characters without following a non-permitted one.

I know. That fact does not contradict what I said AFAICT. Both Tokenizer methods require, in those method callers, the same careful treatment of partial values.

Reply:
> its callers need the same careful treatment of partial values.

Note how BWS is defined:  `BWS = *( SP / HTAB )`. Zero whitespace is a correct value(which, hence, is not 'partial') and the method returns true in this case as a one would expect.

Reply:
> Do you think we should [rename] them in this PR?

No, I think it is best to rename otherwise unchanged methods in a dedicated followup PR.